lib/src/builder/generation/declaration_parsing.dart

```diff
@@ -282,6 +282,34 @@ class ParsedDeclarations {
       );
     }
 
+    // Ensure that Component2 declarations do not use legacy lifecycle methods.
+
+    if (declarationMap[key_component2].isNotEmpty) {
+      final firstComponent2Member = declarationMap[key_component2].first;
+      if (firstComponent2Member is ClassDeclaration) {
+        Map<String, String> legacyLifecycleMethodsMap = {
+          'componentWillReceiveProps': 'Use getDerivedStateFromProps instead.',
+          'componentWillMount': 'Use init instead.',
+          'componentWillUpdate': 'Use getSnapshotBeforeUpdate instead.',
+        };
+
+        legacyLifecycleMethodsMap.forEach((methodName, helpMessage) {
+          final method = firstComponent2Member.getMethod(methodName);
+
+          if (method != null) {
+            error('''Error within ${firstComponent2Member.name.name}.
```

Comment:
This is my bad @joebingham-wk ... but with `'''` comments, in order for the indentation of the source to not influence the output, you have to wrap the string block in `unindent()`, which you can import from `import 'package:over_react/src/util/string_util.dart';`.

Otherwise... the error output in console will look like this...

![image](https://user-images.githubusercontent.com/1750797/58637236-ee8e5c80-82a6-11e9-97c3-12d2af6d57cf.png)

Reply:
No problem - I've worked with comment blocks being logged out before and should have caught that. Thank you for pointing it out again!

Reply:
@aaronlademann-wf that's done! Also, I updated the error message to use `method.name` to keep the message cleaner and not include the method annotations / body. Let me know if the other way was better though